--- include/cstdlib
+++ include/cstdlib
@@ -71,9 +71,9 @@
 
 // Need to ensure this finds the C library's <stdlib.h> not a libstdc++
 // wrapper that might already be installed later in the include search path.
-#define _GLIBCXX_INCLUDE_NEXT_C_HEADERS
-#include_next <stdlib.h>
-#undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS
+
+#include <stdlib.h>
+
 #include <bits/std_abs.h>
 
 // Get rid of those macros defined in <stdlib.h> in lieu of real functions.
